### ISSUE TITLE
Cleanup optimizer tests

### ISSUE THREAD

**Issue by mrks** [hyrise/hyrise#1024]
As discussed in #1022, we should change the *_rule_test.cpp to use `EXPECT_LQP_EQ` instead of simply checking for moved pointers.

> [W]hen implementing optimizer rules you have to make sure the rule doesn't break the LQP. Just testing the presence of nodes can easily miss bugs.
> 
> Second point: If you test the entire LQP, then the reader of the code can be sure the test actually tests everything the rule might or might not do to the LQP. If you just test a selected number of node types, the reader really has to study your test to figure out whether it might miss anything. Which the reader won't do, cause he lazy. Enter error proneness.

**Comment by mrks** [comment]
Also, make sure that tables are only loaded once, not once per test

**Comment by mrks** [comment]
Should change to EXPECT_LQP_EQ:
- [ ] PredicatePlacementRuleTest (~6 tests)
- [ ] PredicateReorderingTest.PredicatesAsRightInput, PredicateReorderingTest.PredicatesWithMultipleOutputs

Should load data only once:
- [ ] PredicatePlacementRuleTest

**Comment by mweisgut** [comment]
>Should load data only once
What data? 

I thought that we could execute the following code only once for the entire test suite (`SetUpTestSuite`):
https://github.com/hyrise/hyrise/blob/a606cf851eb4af300fcd99f49c83fcb6fc7e88b8/src/test/lib/optimizer/strategy/predicate_placement_rule_test.cpp#L25-L50

I think it would be sufficient to load this data only once. The current issue here is that *RuleTest classes are derived from `StrategyBaseTest` which is derived from `BaseTest`. The destructor of `BaseTest` resets Hyrise which is unfortunate for some test cases: In some test suites, e.g. `PredicatePlacementRuleTest`, it is not necessary to reset the storage manager since the test cases don't manipulate it (the storage manager). One reset after the test suite execution in `TearDownTestSuite` would be sufficient.
Concretely in `PredicatePlacementRuleTest`, some test cases create new StoredTableNodes. During the construction, the table is retrieved from the storage manager. If Hyrise and its storage manager are reset after the first test case and the tables are only loaded once (in `SetUpTestSuite`), the second test case cannot access the stored tables anymore.

I haven't checked all test suites in detail yet but I think `PredicatePlacementRuleTest` is not the only test suite where we load unnecessary often. The nature of resetting Hyrise after each test case is an obstacle sometimes.

How about having a `TearDown` and/or `TearDownTestSuite` function in each test suite where Hyrise is reset if required or having a specialized subclass of `BaseTest` that resets Hyrise in its constructor?

**Comment by mrks** [comment]
> How about having a TearDown and/or TearDownTestSuite function in each test suite where Hyrise is reset if required

I would expect someone to forget about this at some point. Hyrise being reset should be the default in all tests.

Can't you simply store the result of `load_table` in the test suite and call `add_table` for each invocation? That won't save us the time of the statistics generation, but at least that of loading the data. I believe that loading tbl files is especially expensive because of how we use AllTypeVariants there.e
